http/src/main/java/io/micronaut/http/annotation/ResponseFilter.java

```diff
@@ -14,12 +20,41 @@
 
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
+/**
+ * Method annotation for a request filter. A response filter is called after a request has been
+ * sent and the response received. Possible parameter types are:
+ *
+ * <ul>
+ *     <li>{@link HttpRequest} or {@link MutableHttpRequest}, to access the request</li>
+ *     <li>{@link HttpResponse} or {@link MutableHttpResponse}, to access the response</li>
+ *     <li>A {@link Throwable} type, to handle an error. If there is an error in the downstream
+ *     filters, it is processed by the upstream response filters. Any response filter that does not
+ *     declare a {@link Throwable} parameter of a matching type is skipped. If instead there is no
+ *     downstream error, those response filters <i>with</i> a {@link Throwable} parameter are
+ *     skipped, unless the parameter is {@link io.micronaut.core.annotation.Nullable}. <b>Note that
+ *     for server filter execution, exceptions are transformed into non-exceptional responses with
+ *     an error status code, between each filter.</b></li>
+ * </ul>
+ *
+ * The return value may be:
+ *
+ * <ul>
+ *     <li>{@code void} to immediately continue execution</li>
+ *     <li>An updated {@link HttpResponse}</li>
+ *     <li>A {@link Publisher} (or other reactive type) that produces any of these return types, to
+ *     delay further execution</li>
+ * </ul>
```

Comment:
We need compilation time checking for these restrictions

Reply:
@graemerocher we don't have a processor for the http module atm, should i create one for this?

Reply:
ive implemented a validator for this, but it would need a new module because if it went into core-processor there would be a circular dependency. im gonna stash this for now since graeme is on vacation